ecosystem/purescript-0.14-migration-guide.md

```diff
@@ -0,0 +1,622 @@
+singleton k a <> singleton k b == singleton k (a <> b)
+```
+
+Code using these instances will now append in the correct order.
+
+**To fix:**
+
+Most users should not change their code and will now be using the correct instance. However, if you were intentionally relying on the behavior of this instance then you can recover it by writing a newtype around `Object` and implementing this newtype instance as desired. For example:
+
+```purs
+newtype Reversed v = Reversed (Object v)
+
+instance semigroupReversed :: Semigroup v => Semigroup (Reversed v) where
+  append (Reversed l) (Reversed r) = Reversed (r <> l)
+```
+
+#### The `Alt` instance for `ZipList` has changed.
+
+Implemented in [purescript-lists#150](https://github.com/purescript/purescript-lists/pull/150).
+
+The previous `Alt` instance for `ZipList` relied on a simple append, which doesn't satisfy the `Alternative` distributivity law:
+
+```purs
+(f <|> g) <*> x == (f <*> x) <|> (g <*> x)
+```
+
+Using an instance implemented with `append`, the left-hand side will be a list of length `min(length f + length g, x)`, whereas the right-hand side will be a list of length `min(length f, length x) + min(length g, length x)`. For example, given:
+
+```purs
+f = g = ZipList ((_ + 1) : Nil)
+x = ZipList (0 : Nil)
+```
+
+we get the left-hand side as `ZipList (1 : Nil)` but the right-hand side `ZipList (1 : 1 : Nil)`.
+
+**To fix:**
+
+Most users will not need to change their code, as this is a bug fix for incorrect behavior. However, if you were previously relying on the specific behavior of this instance, then you can create your own newtype around `List` which defines an `Alt` instance with `append`:
+
+```purs
+newtype MyZipList = MyZipList List
+
+instance altMyZipList :: Alt MyZipList where
+  alt = append
+```
+
+#### The `Record.Builder.merge` and `Record.Builder.union` functions have changed.
+
+Implemented in [purescript-record#73](https://github.com/purescript/purescript-record/pull/73) with additional discussion in [purescript-record#55](https://github.com/purescript/purescript-record/issues/55).
+
+The `Record.Builder.merge` and `Record.Builder.union` functions have changed to behave more like `Record.merge` and `Record.union`, in that fields from the argument will override those of the record being built in the case of overlaps. This is a breaking change because the result types may change -- for example, this:
+
+```purs
+Builder.build (Builder.merge { x: 1, y: "y" }) { y: 2, z: true }
+```
+
+will now produce the type:
+
+```purs
+{ x :: Int, y :: String, z :: Boolean }
+```
+
+where previously it would produce the type:
+
+```purs
+{ x :: Int, y :: Int, z :: Boolean }
+```
+
+**To fix:**
+
+If you would like to use the new behavior, then your code doesn't need to change. Otherwise, you can use the new `Record.Builder.flip` function to flip the arguments and recover the old behavior.
+
+```purs
+Builder.build (Builder.flip Builder.merge ...)
```

Comment:
I think we can reuse the arguments of the previous example, and add a type annotation:

```suggestion
Builder.build (Builder.flip Builder.merge { x: 1, y: "y" }) { y: 2, z: true }
  :: { x :: Int, y :: Int, z :: Boolean }
```